app/src/app/api/v0/city/invite/route.ts

```diff
@@ -9,70 +9,58 @@ import jwt from "jsonwebtoken";
 import { sendEmail } from "@/lib/email";
 import { render } from "@react-email/components";
 import InviteUserTemplate from "@/lib/emails/InviteUserTemplate";
+import UserService from "@/backend/UserService";
 
-export const POST = apiHandler(
-  async (
-    _req: NextRequest,
-    context: { session?: Session; params: Record<string, string> },
-  ) => {
-    const { params, session } = context;
+export const POST = apiHandler(async (_req, { params, session }) => {
+  const body = createUserInvite.parse(await _req.json());
+  const city = await UserService.findUserCity(body.cityId, session);
 
-    const body = createUserInvite.parse(await _req.json());
+  const cityData = await db.models.City.findOne({
+    where: { cityId: city.cityId },
+  });
 
-    if (!session) throw new createHttpError.Unauthorized("Unauthorized");
-    const user = await db.models.User.findOne({
-      where: { userId: body.userId },
-    });
+  if (!cityData) {
+    throw new createHttpError.NotFound("City not found");
+  }
 
-    if (!user) {
-      throw new createHttpError.NotFound("User not found");
-    }
+  if (!process.env.VERIFICATION_TOKEN_SECRET) {
+    console.error("Need to assign RESET_TOKEN_SECRET in env!");
+    throw createHttpError.InternalServerError("Configuration error");
+  }
 
-    const city = await db.models.City.findOne({
-      where: { locode: body.locode },
-    });
-    if (!city) {
-      throw new createHttpError.NotFound("City not found");
-    }
+  const invitationCode = jwt.sign(
+    { email: session?.user.email },
+    process.env.VERIFICATION_TOKEN_SECRET,
+    {
+      expiresIn: "1h",
+    },
+  );
 
-    if (!process.env.VERIFICATION_TOKEN_SECRET) {
-      console.error("Need to assign RESET_TOKEN_SECRET in env!");
-      throw createHttpError.InternalServerError("Configuration error");
-    }
+  const invite = await db.models.CityInvite.create({
+    id: randomUUID(),
+    ...body,
+  });
 
-    const invitationCode = jwt.sign(
-      { email: user.email },
-      process.env.VERIFICATION_TOKEN_SECRET,
-      {
-        expiresIn: "1h",
-      },
-    );
+  if (!invite) {
+    throw new createHttpError.BadRequest("Something went wrong");
+  }
+  const host = process.env.HOST ?? "http://localhost:3000";
+  const sendInvite = await sendEmail({
+    to: body.email!,
+    subject: "Test Invite",
+    html: render(
+      InviteUserTemplate({
+        url: `${host}/api/v0/city/invite/${invite.id}?token=${invitationCode}&email=${body.email}`,
+        user: { email: body.email, name: body.name },
+        city,
+        invitee: { name: session?.user.name!, email: session?.user.email! },
```

Comment:
Why is the invitee the current user? Shouldn't it be the user that's being invited?
Otherwise rename this variable to `inviter` or `invitingUser` (both here and in the email template)